packages/react/ds/src/card/card.tsx

```diff
@@ -1,61 +1,136 @@
 import { Icon, IconPropTypes } from '../icon/icon.js';
-import { Link } from '../link/link.js';
+import { Link, LinkProps } from '../link/link.js';
 import { Paragraph } from '../paragraph/paragraph.js';
+import { Tag, TagProps } from '../tag/tag.js';
 
-type Action = {
-  href: string;
-  text: string;
-};
+type Action =
+  | (ButtonProps & { type: 'button' })
+  | (LinkProps & { type: 'link' });
 
 export type CardProps = {
   type: 'vertical' | 'horizontal';
+  inset?: 'body' | 'full' | 'none';
   title?: string;
+  subTitle?: string;
   href?: string;
   img?: string;
+  tag?: TagProps;
   icon?: IconPropTypes;
   content?: string;
-  actions?: Action[];
+  action?: Action;
 };
 
-export const Card = (props: CardProps) => {
-  const { type = 'vertical', title, img, icon, content, actions, href } = props;
-  const layoutClasses = {
-    vertical: 'gi-card gi-card-vertical',
-    horizontal: 'gi-card gi-card-horizontal',
-  };
+// Memoized truncation utility
+const useTruncatedText = (text: string | undefined, maxLength: number) => {
+  return useMemo(() => {
+    if (!text) {
+      return '';
+    }
+    return text.length > maxLength ? `${text.slice(0, maxLength)}...` : text;
+  }, [text, maxLength]);
+};
 
-  return (
-    <div className={layoutClasses[type]}>
-      {img && !icon && (
+const CONTENT_MAX_LENGTH = 120;
+const TITLE_MAX_LENGTH = 60;
+
+export const Card = ({
+  type = 'vertical',
+  title,
+  inset = 'none',
+  subTitle,
+  img,
+  icon,
+  content,
+  action,
+  href,
+  tag,
+}: CardProps) => {
+  const truncatedContent = useTruncatedText(content, CONTENT_MAX_LENGTH);
+  const truncatedTitle = useTruncatedText(title, TITLE_MAX_LENGTH);
+  const truncatedSubTitle = useTruncatedText(subTitle, TITLE_MAX_LENGTH);
+
+  const cardClasses = useMemo(() => {
+    const insetClass = `gi-card-inset-${inset}`;
+    return `gi-card gi-card-${type} ${insetClass}`;
+  }, [type, inset]);
+
+  const renderMedia = () => {
+    if (img && !icon) {
+      return (
         <div className="gi-card-image">
           <a href={href}>
             <img src={img} alt={title} />
           </a>
         </div>
-      )}
-      {icon && (
+      );
+    }
+
+    if (icon) {
+      return (
         <div className="gi-card-icon">
           <a href={href}>
             <Icon {...icon} />
           </a>
         </div>
-      )}
-      <div className="gi-card-content">
-        <div className="gi-card-heading">
-          <Heading as="h5" customClasses="!gi-my-0">
-            {href ? <Link href={href}>{title}</Link> : title}
-          </Heading>
-        </div>
-        <div className="gi-card-paragraph">
-          <Paragraph size="sm">{content}</Paragraph>
+      );
+    }
+
+    return null;
+  };
+
+  const renderTitle = () => {
+    const titleContent = href ? (
+      <Link href={href}>{truncatedTitle}</Link>
+    ) : (
+      truncatedTitle
+    );
+
+    return (
+      <Heading
+        as="h5"
```

Comment:
You shouldn't use an `h5` here; instead, use a `div` with the same style as an `h5`.